Update how to remove trusted clusters. (#4358)
gravitational/teleport: docs/4.3/trustedclusters.md

experience looks like this:

```bsh
# login using the root "root" cluster credentials:
$ tsh login --proxy=main.example.com

# SSH into some host inside the "root" cluster:
$ tsh ssh host

# SSH into the host located in another cluster called "leaf"

> # The connection is established through main.example.com:

Reviewer: ```suggestion
# The connection is established through root.example.com:
```

    version does not support SSH roles.

When a _trusting_ cluster "leaf" from the diagram above establishes trust with
the _trusted_ cluster "root", it needs a way to configure which users from
"root" should be allowed in and what permissions should they have. Teleport
Enterprise uses _role mapping_ to achieve this.

Consider the following:

* Both clusters "root" and "leaf" have their own locally defined roles.
* Every user in Teleport Enterprise is assigned a role.
* When creating a _trusted cluster_ resource, the administrator of "leaf" must
  define how roles from "root" map to roles on "leaf".
* To update role map for an existing _trusted cluster_ delete and re-create the _trusted cluster_ with the updated role map

### Example

Lets make a few assumptions for this example:

* The cluster "root" has two roles: _user_ for regular users and _admin_ for
  local administrators.

* We want administrators from "root" (but not regular users!) to have
  restricted access to "leaf". We want to deny them access to machines
  with "environment=production" label.

First, we need to create a special role for main users on "leaf":

```yaml

> # save this into main-user-role.yaml on the east cluster and execute:

Reviewer: ```suggestion
# save this into root-user-role.yaml on the leaf cluster and execute:
```

$ tctl create main-user-updated-role.yaml
```

## Using Trusted Clusters

Now an admin from the main cluster can now see and access the "leaf" cluster:

```bsh

> # login into the main cluster:

Reviewer: ```suggestion
# log into the root cluster:
```

```bsh
# SSH into any node in "leaf":
$ tsh ssh --cluster=leaf root@db1.leaf
```

!!! tip "Note"

    Trusted clusters work only one way. So, in the example above users from "leaf"
    cannot see or connect to the nodes in "root".

### Disabling Trust

To temporarily disable trust between clusters, i.e. to disconnect the "leaf"
cluster from "root", edit the YAML definition of the trusted cluster resource
and set `enabled` to "false", then update it:

```bsh
$ tctl create --force cluster.yaml
```

### Remove Leaf Cluster relationship from both sides

Once established, to fully remove a trust relationship between two clusters, do
the following:

- Remove the relationship from the leaf cluster: `tctl rm tc/root.example.com`.
- Remove the relationship from the root cluster: `tctl rm lc/leaf.example.com`.

### Remove Leaf Cluster relationship from the root

Remove the relationship from the root cluster: `tctl rm rc/leaf.example.com`.

!!! Note

    The `leaf.example.com` cluster will continue to try and ping the root cluster,
    but will not be able to connect. To re-establish the trusted cluster relationship,
    the trusted cluster has to be create again from the leaf cluster.

> ### Remote Leaf Cluster relationship from the Leaf.

Reviewer: ```suggestion
### Remove Leaf Cluster relationship from the leaf
```

trust between two clusters:

> * **HTTPS configuration**: when the main cluster uses a self-signed or invalid HTTPS certificate.

Reviewer: ```suggestion
* **HTTPS configuration**: when the root cluster uses a self-signed or invalid HTTPS certificate.
```

* **Connectivity problems**: when a trusting cluster "leaf" does not show up in
  `tsh clusters` output on "root".
* **Access problems**: when users from "root" get "access denied" error messages
  trying to connect to nodes on "leaf".

### HTTPS configuration

> If the `web_proxy_addr` endpoint of the main cluster uses a self-signed or invalid HTTPS certificate,

Reviewer: ```suggestion
If the `web_proxy_addr` endpoint of the root cluster uses a self-signed or invalid HTTPS certificate,
```

you will get an error: "the trusted cluster uses misconfigured HTTP/TLS certificate". For ease of

> testing the teleport daemon of "leaf" can be started with  `--insecure` CLI flag to accept

Reviewer: ```suggestion
testing, the teleport daemon on "leaf" can be started with the `--insecure` CLI flag to accept
```
